client/src/Todo.js

```diff
@@ -1,25 +1,74 @@
-import React from "react";
+import React, { useState } from "react";
 import { useTodo } from "./TodoProvider";
 import IconButton from "@material-ui/core/IconButton";
+import EditIcon from "@material-ui/icons/Edit";
 import DeleteIcon from "@material-ui/icons/Delete";
+import SaveIcon from "@material-ui/icons/Save";
+import "./styles.css";
 
-export default function Todo({ id, description, complete }) {
-  const { setStatusTodo, deleteTodo } = useTodo();
+export default function Todo({ id, description, completed }) {
+  const { setStatusTodo, deleteSingleTodo, updateTodoDescription } = useTodo();
 
   const checkTodo = (e) => setStatusTodo(id, e.target.checked);
 
+  const [editTodoId, setEditTodoId] = useState(null);
+  const [editedTodoDescription, setEditedTodoDescription] = useState("");
+
+  const handleEdit = (id, currentDescription) => {
+    setEditTodoId(id);
+    setEditedTodoDescription(currentDescription);
+  };
+
+  const handleSave = (id) => {
+    updateTodoDescription(id, editedTodoDescription);
+    setEditTodoId(null);
+    setEditedTodoDescription("");
+  };
+
+  const handleDelete = (id) => {
+    deleteSingleTodo(id);
+  };
+
   return (
     <tr>
       <td>
-        <input type="checkbox" onChange={checkTodo} checked={complete} />
+        <input type="checkbox" onChange={checkTodo} checked={completed} />
       </td>
-      <td>
-        <span className={complete ? "todo-done" : ""}>{description}</span>
-      </td>
-      <td>
-        <IconButton className="close" onClick={() => deleteTodo(id)}>
-          <DeleteIcon />
-        </IconButton>
+      <td key={id}>
+        {editTodoId === id ? (
```

Comment:
This line can be different/simpler if you just have a boolean variable, e.g.  "isEditMode"

Reply:
changed it?